dataservice/api/common/schemas.py

```diff
@@ -156,6 +154,42 @@ class PaginatedSchema(Schema):
     return PaginatedSchema
 
 
+def filter_schema_factory(model_filter_schema_cls):
+    """
+    Create an instance of a model's filter schema
+
+    Remove schema attributes that are not applicable to filters (_links)
+    Allow partially populated schema
+    Validate with strict=True - reuse model schema's validators
+    """
+    # Exclude fields
+    exclude = ('_links', )
+    if hasattr(model_filter_schema_cls.Meta, 'exclude'):
+        exclude += model_filter_schema_cls.Meta.exclude
+
+    return model_filter_schema_cls(strict=True,
+                                   partial=True,
+                                   exclude=exclude)
+
+
+class FilterSchemaMixin(ma.ModelSchema):
+    """
+    Filter schema mixin inherited by all model filter schemas
+    """
+    study_id = fields.Str()
+
+    class Meta:
+        dump_only = ()
+
+    @validates('study_id')
+    def valid(self, value):
+        validate_kf_id('SD', value)
+
+    @post_load
+    def make_instance(self, data):
```

Comment:
What does this do?

Reply:
When you use use_args from webargs, it does a schema load. So if you want it to return the dict instead of the object after the load, you just return the data.

Reply:
Isn't `post_load` just a hook? Or does it override something?

Reply:
So it is a little obscure. `post_load` is a hook, but `make_instance` is also a method in the base class which does the deserialization from the data dict to the model. So i'm just having it override `make_instance` to return the dict, but doing this after load so that validation still happens. Seems confusing to me bc you would think that the deserialization has already happened post load, but this works. I found the suggestion here on webargs github:

Idk if there is a better way to do this?

https://github.com/sloria/webargs/issues/126